ui/js/controllers/logviewer.js

```diff
@@ -33,13 +32,58 @@ logViewerApp.controller('LogviewerCtrl', [
         $scope.currentLineNumber = 0;
         $scope.highestLine = 0;
         $scope.showSuccessful = true;
-
+        $scope.willScroll = false;
         $scope.$watch('artifact', function () {
             if (!$scope.artifact) {
                 return;
             }
             $scope.showSuccessful = !$scope.hasFailedSteps();
         });
+        $scope.$watch('[selectedBegin, selectedEnd]', function(newVal, oldVal) {
+            var newHash = (newVal[0] == newVal[1])? newVal[0] : newVal[0] + "-L" + newVal[1];
```

Comment:
This needs a space before the ? operator if possible.
`newVal[1]) ?`

Maybe you can test for strict type equivalency also via `===` if that makes sense and is possible.